cookbook/profiler/data_collector.rst

```diff
@@ -46,106 +46,160 @@ access to in local properties.
+    // src/AppBundle/DataCollector/MyCollector.php
+    class MyCollector extends DataCollector
     {
         public function collect(Request $request, Response $response, \Exception $exception = null)
         {
             $this->data = array(
-                'memory' => memory_get_peak_usage(true),
+                'variable' => 'value',
             );
         }
 
-        public function getMemory()
+        public function getVariable()
         {
-            return $this->data['memory'];
+            return $this->data['variable'];
         }
 
         public function getName()
         {
-            return 'memory';
+            return 'app.my_collector';
         }
     }
 
 .. _data_collector_tag:
 
-Enabling custom Data Collectors
+Enabling Custom Data Collectors
 -------------------------------
 
-To enable a data collector, add it as a regular service in one of your
-configuration, and tag it with ``data_collector``:
+To enable a data collector, define it as a regular service in your application
+and tag it with ``data_collector``:
 
 .. configuration-block::
 
     .. code-block:: yaml
 
+        # app/config/services.yml
         services:
-            data_collector.your_collector_name:
-                class: Fully\Qualified\Collector\Class\Name
+            app.my_collector:
+                class: AppBundle\DataCollector\MyCollector
                 tags:
                     - { name: data_collector }
 
     .. code-block:: xml
 
-        <service id="data_collector.your_collector_name" class="Fully\Qualified\Collector\Class\Name">
+        <!-- app/config/services.xml -->
+        <service id="app.my_collector" class="AppBundle\DataCollector\MyCollector">
             <tag name="data_collector" />
         </service>
 
     .. code-block:: php
 
+        // app/config/services.php
         $container
-            ->register('data_collector.your_collector_name', 'Fully\Qualified\Collector\Class\Name')
+            ->register('app.my_collector', 'AppBundle\DataCollector\MyCollector')
             ->addTag('data_collector')
         ;
 
 Adding Web Profiler Templates
 -----------------------------
 
-When you want to display the data collected by your data collector in the web
-debug toolbar or the web profiler, you will need to create a Twig template. The
-following example can help you get started:
+The information collected by your data collector can be displayed both in the
+web debug toolbar and in the web profiler. To do so, you need to create a Twig
+template that includes some specific blocks.
+
+In the simplest case, you just want to display the information in the toolbar
+without providing a profiler panel. This requires to define the ``toolbar``
+block and set the value of two variables called ``icon`` and ``text``:
 
 .. code-block:: jinja
 
-    {% extends 'WebProfilerBundle:Profiler:layout.html.twig' %}
+    {% extends '@WebProfiler/Profiler/layout.html.twig' %}
```

Comment:
The WebProfilerBundle uses the native Twig feature everywhere, to be able to use it outside the fullstack framework (in Silex or Drupal for instance).

On a side note, using the bundle notation here would mean that Twig has to compile the template again under a different cache key.